cdap-watchdog/src/main/java/co/cask/cdap/logging/framework/distributed/DistributedLogFramework.java

```diff
@@ -91,12 +94,14 @@ protected Service createService(Set<Integer> partitions) {
 
       long bufferSize = getBufferSize(pipelineCount, cConf, partitions.size());
       final String topic = cConf.get(Constants.Logging.KAFKA_TOPIC);
-
+      final LogsMetricsContext logsMetricsContext = new LogsMetricsContext(context.getMetricsContext(),
+                                                                           context.getName(), context.getInstanceId());
       final KafkaPipelineConfig config = new KafkaPipelineConfig(
         topic, partitions, bufferSize,
         cConf.getLong(Constants.Logging.PIPELINE_EVENT_DELAY_MS),
         cConf.getInt(Constants.Logging.PIPELINE_KAFKA_FETCH_SIZE),
-        cConf.getLong(Constants.Logging.PIPELINE_CHECKPOINT_INTERVAL_MS)
+        cConf.getLong(Constants.Logging.PIPELINE_CHECKPOINT_INTERVAL_MS),
+        logsMetricsContext
```

Comment:
The config object is just for configuration, shouldn't include the MetricsContext. The MetricsContext should be exposed/implemented by the `LogProcessorPipelineContext` class instead.